### PR TITLE
Fix #5036: quickstart: Typing Ctrl-U clears the whole of line

### LINKED ISSUE #5036: Typing Control-U in sphinx-quickstart clears the line
To reproduce, start `sphinx-quickstart test`, start typing something for the first answer, and press `Control-U`. The entire question line is cleared, rather than just what was typed.

### Environment info
- OS: Mac
- Python version: 3.5
- Sphinx version: 1.7.4

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Bugfix

### Purpose
- refs: #5036

### CODE REVIEW

**Comment by shimizukawa** [comment]
image

Unfortunately, ti seems that Windows can't handle `\1` and `\2` esc sequence correctly.

File: sphinx/cmd/quickstart.py [hunk 1]
```diff
@@ -82,8 +82,7 @@
 # function to get input from terminal -- overridden by the test suite
 def term_input(prompt):
     # type: (unicode) -> unicode
-    print(prompt, end='')
-    return input('')
+    return input(prompt)
```
Comment (shimizukawa) [inline sphinx/cmd/quickstart.py]: I've found a problem on Windows.

image

```
C:\Users\taka>python
Python 3.6.2 (v3.6.2:5fd33b5, Jul  8 2017, 04:14:34) [MSC v.1900 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from sphinx.util.console import colorize
>>> colorize('yellow', 'Hello', True)
'\x01\x1b[33;01m\x02Hello\x01\x1b[39;49;00m\x02'
>>> colorize('yellow', 'Hello', False)
'\x1b[33;01mHello\x1b[39;49;00m'
>>> from sphinx.util.console import color_terminal
>>> color_terminal()
True
>>> input(colorize('yellow', 'Hello', False))
[33;01mHello[39;49;00m False
' False'
>>> input(colorize('yellow', 'Hello', True))
[33;01mHello[39;49;00m True
' True'
>>> print(colorize('yellow', 'Hello', False))
Hello
>>> print(colorize('yellow', 'Hello', True))
Hello
```

I guess `input()` doesn't handle escape sequence correctly.
Previous code (print + input) works fine.

Comment (shimizukawa) [reply sphinx/cmd/quickstart.py]: https://github.com/tk0miya/sphinx/commit/678cc3d50363e927ab904b61e9d7eaee04f4e965 separate `term_input` into `print` and `input` that shipped with Sphinx-1.5.3.

Before sphinx-1.5.3, Windows can't print escape sequence with `input` correctly too.
image

I think we should keep `print` + `input` code for windows environment.

**Comment by tk0miya** [comment]
@shimizukawa Thank you for investigation. I just updated to skip windows. Could you test this again please?

**Comment by shimizukawa** [comment]
No different from previous.
image

It seems `input('<escape sequence>')` doesn't work correctly.
According to the report #3406, if windows user install `pyreadline`, the environment accept the escape sequence at `input()`. However, I feel that it is safe to display it by using `print(message)` and `input('')` as usual for windows users.

**Comment by tk0miya** [comment]
@shimizukawa Oh, I'm very sorry. My patch was wrong. I updated it again.
Could you check this again please?

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Comment by shimizukawa** [comment]
It works well in my Win10 PowerShell!!

image

**Comment by tk0miya** [comment]
Thank you for confirming!

**Comment by tk0miya** [comment]
Merged!

**Pull request merged**